kadai-adapter/src/main/java/io/kadai/adapter/monitoring/HealthCheck.java

```diff
@@ -0,0 +1,199 @@
+import org.springframework.core.ParameterizedTypeReference;
+import org.springframework.core.env.Environment;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Component;
+import org.springframework.web.client.RestTemplate;
+
+@Component("externalServices")
+public class HealthCheck implements HealthIndicator {
+
+  private final RestTemplate restTemplate;
+
+  @Autowired private Environment env;
+
+  @Autowired
+  public HealthCheck(RestTemplate restTemplate) {
+    this.restTemplate = restTemplate;
+  }
+
+  @Override
+  public Health health() {
+    Health.Builder camundaHealth = Health.up();
+    Health.Builder outboxHealth = Health.up();
+    Health.Builder kadaiHealth = Health.up();
+    try {
+      ResponseEntity<EngineInfoRepresentationModel[]> camundaResponse = pingCamundaRest();
+      EngineInfoRepresentationModel[] engines = camundaResponse.getBody();
+      if (engines != null
+          && engines.length > 0
+          && camundaResponse.getStatusCode().equals(HttpStatus.OK)) {
+        camundaHealth
+            .withDetail("Camunda Engines", engines)
+            .status(Health.up().build().getStatus());
+      } else {
+        camundaHealth
+            .withDetail("Camunda Engine", "No engines found")
+            .status(Health.down().build().getStatus());
+      }
+    } catch (Exception e) {
+      camundaHealth
+          .withDetail("Camunda Engine Error", e.getMessage())
+          .status(Health.down().build().getStatus());
+    }
+
+    try {
+      ResponseEntity<EventCountRepresentationModel> outboxResponse =
+          pingOutBoxRestGetNumberFailedEvents();
+      outboxHealth
+          .withDetail("Outbox Service", outboxResponse.getBody())
+          .status(Health.up().build().getStatus());
+    } catch (Exception e) {
+      outboxHealth
+          .withDetail("Outbox Service Error", e.getMessage())
+          .status(Health.down().build().getStatus());
+    }
+
+    try {
+      ResponseEntity<VersionInfoRepresentationModel> kadaiResponse =
+          pingKadaiGetCurrentSchemaVersion();
+      kadaiHealth
+          .withDetail("Kadai Service", kadaiResponse.getBody())
+          .status(Health.up().build().getStatus());
+    } catch (Exception e) {
+      kadaiHealth
+          .withDetail("Kadai Service Error", e.getMessage())
+          .status(Health.down().build().getStatus());
+    }
+
+    return Health.status("External Services Status")
+        .withDetail("Camunda Health", camundaHealth.build())
+        .withDetail("Outbox Health", outboxHealth.build())
+        .withDetail("Kadai Health", kadaiHealth.build())
+        .build();
+  }
+
+  private ResponseEntity<EngineInfoRepresentationModel[]> pingCamundaRest() throws Exception {
```

Comment:
Can't we extract just one method for `pingCamundaRest`, `pingOutBoxRestGetNumberFailedEvents` and `pingKadaiGetCurrentSchemaVersion`?

Reply:
What do you mean?